packages/fuels-contract/src/contract.rs

```diff
@@ -173,7 +198,8 @@ impl Contract {
         storage_slots: Vec<StorageSlot>,
         salt: Salt,
     ) -> Result<ContractId, Error> {
-        let mut compiled_contract = Contract::load_sway_contract_with_salt(binary_filepath, salt)?;
+        let mut compiled_contract =
+            Contract::load_sway_contract_with_salt(binary_filepath, None, salt)?;
```

Comment:
In addition to the above comment on `deploy`, it is confusing to have `deploy` do storage with manual/automatic merging while `deploy_with_parameters` only does manual storage.